### PR TITLE
HEEDLS-703 - Migration to add question comparison columns to Competen…

### PR DESCRIPTION
**Pull request by DanBloxham-sw**
…cyResourceAssessmentQuestionParameters

### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-703

### Description
New migration adding two new columns to the ComptencyResourceAssessmentQuestionParameters that will be used in the Recommendation Score generation for learning resources.

### Screenshots
N/A

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [x] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/19875758087/Recommendation+Score
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Data.Migrations/202112091009_AddColumnsToCompetencyResourceAssessmentQuestionParameters.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        }
+
+        public override void Down()
+        {
+            Delete.ForeignKey(
+                    "FK_CompetencyResourceAssessmentQuestionParameters_RelevanceAssessmentQuestionID_AssessmentQuestions_ID"
+                )
+                .OnTable("CompetencyResourceAssessmentQuestionParameters");
+            Delete.Column("RelevanceAssessmentQuestionID").FromTable("CompetencyResourceAssessmentQuestionParameters");
+            Delete.Column("CompareToRoleRequirements ").FromTable("CompetencyResourceAssessmentQuestionParameters");
```
Comment (livzorn) [inline DigitalLearningSolutions.Data.Migrations/202112091009_AddColumnsToCompetencyResourceAssessmentQuestionParameters.cs:23]: Looks like an extra space after CompareToRoleRequirements

**Review by livzorn — CHANGES_REQUESTED** [review]
Just one extra space, not sure if that affects anything?

File: DigitalLearningSolutions.Data.Migrations/202112091009_AddColumnsToCompetencyResourceAssessmentQuestionParameters.cs (hunk 1 above)
Comment (DanBloxham-sw) [reply DigitalLearningSolutions.Data.Migrations/202112091009_AddColumnsToCompetencyResourceAssessmentQuestionParameters.cs]: Good spot. I tested the Down migration while it had the space, and the column was removed fine, so it looks like the Migrator is clever enough to trim these. I've removed the space now.

**Review by livzorn — APPROVED** [review]

**Review by AlexJacksonDS — COMMENTED** [review]
In theory this looks fine, I'm not particularly keen on this column name though "RelevanceAssessmentQuestionID". Do we actually need it to be called this, or should it really be called "RelevantAssessmentQuestionID"?

**Comment by DanBloxham-sw** [comment]
> In theory this looks fine, I'm not particularly keen on this column name though "RelevanceAssessmentQuestionID". Do we actually need it to be called this, or should it really be called "RelevantAssessmentQuestionID"?

I agree that it is not the nicest name, but I think it is the most appropriate. These columns are used in the context of generating the recommendation score for a particular learning resource. In that, we recommend items that are more useful to the user, which is partially determined by their answers in self assessment questions concerning related competencies. 

These questions should follow along a pattern of "How confident are you in {dealing with x}" followed by "How well trained do you think you need to be in {dealing with x}". These two types of questions are effectively "Confidence" and "Relevance" questions respectively - i.e., how confident are you in X, and how relevant do you think it actually is to you and your job. 

So, in a case where we look at these questions, we need to know which question is which, so we can look at the user's answers and alter the recommendations score from them. This table already has an "`AssessmentQuestionID`" that points to the Confidence question. The new "`RelevanceAssessmentQuestionID`" points to the Relevance question. So a name like "`RelevantAssessmentQuestionID`" might be subtly confusing - what makes a question the relevant one? I think `RelevanceAssessmentQuestionID` is more "off" sounding in natural English, which hopefully keys anyone in that it is more specific than just a relevant question.

While it would be good to rename the old `AssessmentQuestionID` as "`ConfidenceAssessmentQuestionID`", it is not guaranteed to be a Confidence question - the managers of self assessments can set them up differently, and choose different ways to assess delegates. But, the new `RelevanceAssessmentQuestionID` column should only be populated when confidence/relevance questions are used, and in that case the `AssessmentQuestionID` is going to be a confidence question.

Hopefully that makes sense, but I'm happy to clear anything up further if it doesn't.

**Review by AlexJacksonDS — APPROVED** [review]

**Pull request merged**